src/mono/browser/runtime/http.ts

```diff
@@ -45,19 +52,27 @@ export function http_wasm_supports_streaming_response(): boolean {
     return typeof Response !== "undefined" && "body" in Response.prototype && typeof ReadableStream === "function";
 }
 
-export function http_wasm_create_abort_controler(): AbortController {
+export function http_wasm_create_controller(): HttpController {
     verifyEnvironment();
-    return new AbortController();
+    assert_js_interop();
+    const controller: Partial<HttpController> = {
```

Comment:
Since everything except the `abortController` is optional now, do you still need these TS castings (Partial / as)?